unittest/test_util_Tokenizer.cpp

```diff
@@ -142,4 +146,121 @@ TEST_CASE("util::Tokenizer")
       CHECK(s[1] == "a");
     }
   }
+
+  SUBCASE("include empty and delimiter")
+  {
+    const auto split = split_func(Mode::include_empty, IncludeDelimiter::yes);
+    {
+      const auto s = split("", "/");
+      REQUIRE(s.size() == 1);
+      CHECK(s[0] == "");
+    }
+    {
+      const auto s = split("/", "/");
+      REQUIRE(s.size() == 1);
```

Comment:
This feels surprising to me. I thought that `IncludeDelimiter::yes` should just affect each token, not how tokens are split. Put differently: For every pair of input and delimiters strings, I would expect the number of tokens to be equal with `IncludeDelimiter::yes` and `IncludeDelimiter::no` in all split modes. Does this make sense to you?

Reply:
But it is! Did I miss anything?

Reply:
With `Mode::include_empty` and `IncludeDelimiter::no`, splitting `"/"` with delimiters `"/"` returns 2 tokens:

https://github.com/ccache/ccache/blob/5486a6809ae39d15e8e5ec88dce977778358ba14/unittest/test_util_Tokenizer.cpp#L46-L47

With `Mode::include_empty` and `IncludeDelimiter::yes`, splitting `"/"` with delimiters `"/"` returns 1 token:

https://github.com/ccache/ccache/blob/5486a6809ae39d15e8e5ec88dce977778358ba14/unittest/test_util_Tokenizer.cpp#L159-L160

Reply:
Ok, you're right of course. I'll try to debug it.

Reply:
Done